Google.Solutions.Compute/Iap/IapTunnelingEndpoint.cs

```diff
@@ -0,0 +1,111 @@
+﻿//
+// Copyright 2019 Google LLC
+//
+// Licensed to the Apache Software Foundation (ASF) under one
+// or more contributor license agreements.  See the NOTICE file
+// distributed with this work for additional information
+// regarding copyright ownership.  The ASF licenses this file
+// to you under the Apache License, Version 2.0 (the
+// "License"); you may not use this file except in compliance
+// with the License.  You may obtain a copy of the License at
+// 
+//   http://www.apache.org/licenses/LICENSE-2.0
+// 
+// Unless required by applicable law or agreed to in writing,
+// software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.
+//
+
+using Google.Apis.Auth.OAuth2;
+using Google.Solutions.Compute.Net;
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Net;
+using System.Net.WebSockets;
+using System.Threading;
+using System.Threading.Tasks;
+
+namespace Google.Solutions.Compute.Iap
+{
+    public class IapTunnelingEndpoint : ISshRelayEndpoint
+    {
+        private const string BaseUri = "wss://tunnel.cloudproxy.app/v4/";
+        private const string SubprotocolName = "relay.tunnel.cloudproxy.app";
+        private const string Origin = "bot:iap-tunneler";
+        public const string DefaultNetworkInterface = "nic0";
```

Comment:
could we change this to local interface (eg, lo0)?  if we make this listenening on deault interface, i think external systems can connect ...which i don't think we shoudl allow at all...

Reply:
This is the interface of the remote VM and we have to use nic0 here. 

The `SshRelayListener` which opens the local port on the client computer always binds to 127.0.0.1.